<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NMediaManager::NPreferences::NUi::CTranscodeVideoSettings</class>
 <widget class="QWidget" name="NMediaManager::NPreferences::NUi::CTranscodeVideoSettings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>836</width>
    <height>855</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="transcodeVideo">
     <property name="title">
      <string>Transcode Video?</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QCheckBox" name="onlyTranscodeVideoOnFormatChange">
        <property name="text">
         <string>Only Transcode Video when Format requires it?</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>     (If checked, Video Transcoding will only occur when a format is changed, and will ignore other requirements)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>Hardware Encoding/Decoding</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="1" column="3">
          <widget class="QRadioButton" name="softwareTranscode">
           <property name="text">
            <string>Software</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QRadioButton" name="nvidiaGPUTranscode">
           <property name="text">
            <string>nVidia GPU</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QRadioButton" name="amdGPUTranscode">
           <property name="text">
            <string>AMD GPU</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QRadioButton" name="intelGPUTranscode">
           <property name="text">
            <string>Intel GPU</string>
           </property>
          </widget>
         </item>
         <item row="1" column="4">
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>590</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="2" column="0" colspan="5">
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Hardware Accelerator:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="hwAccel"/>
           </item>
           <item>
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>37</width>
               <height>17</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Output Codec:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="videoCodec"/>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="generateNon4kVideo">
        <property name="text">
         <string>For video with a resolution of 4k or greater, create secondary 1920x1080 video?</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Bitrate Targets (Used for both Secondary Video creation and HEVC Target Bitrate)</string>
        </property>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="1">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>(Higher is Better Quality)</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>4k Video Bitrate:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="NSABUtils::CLineEditWithSuffix" name="average4kBitRate">
           <property name="placeholderText">
            <string>Bitrate in kbps</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>1080p (HD) Video Bitrate:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="NSABUtils::CLineEditWithSuffix" name="averageHDBitRate">
           <property name="maxLength">
            <number>10</number>
           </property>
           <property name="placeholderText">
            <string>Bitrate in kbps</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Sub HD Video Bitrate:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="NSABUtils::CLineEditWithSuffix" name="averageSubHDBitRate">
           <property name="maxLength">
            <number>10</number>
           </property>
           <property name="placeholderText">
            <string>Bitrate in kbps</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Non-Comforming Resolution Divisor:</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QSpinBox" name="nonConformingResolutionRatio">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>10000</number>
           </property>
           <property name="stepType">
            <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
           </property>
          </widget>
         </item>
         <item row="6" column="0" colspan="2">
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Uncompressed Bitrate is calculated then divided by this number, Lower is Better Quality&lt;/p&gt;&lt;p align=&quot;center&quot;&gt;For example: 24bit 1080p with a divisor of 250=&lt;br/&gt;24*1920*1080*60/250 = 11.94 kbit/sec&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="textFormat">
            <enum>Qt::RichText</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="generateLowBitrateVideo">
        <property name="title">
         <string>For video with high bitrates, create secondary lower bitrate video?</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="2">
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string>(If bitrate is reported as greater than Target Bitrate +/- threshhold percentage)</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>Threshold:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <spacer name="horizontalSpacer_9">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="bitrateThreshold">
           <property name="suffix">
            <string>%</string>
           </property>
           <property name="minimum">
            <number>30</number>
           </property>
           <property name="maximum">
            <number>200</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="h265Options">
        <property name="title">
         <string>H.265 (HEVC) Options</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="7" column="0" colspan="3">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QCheckBox" name="useProfile">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Profile:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="profile">
             <item>
              <property name="text">
               <string>main</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main-intra</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>mainstillpicture</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main444-8</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main444-intra</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main444-stillpicture</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main10</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main10-intra</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main422-10</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main422-10-intra</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main444-10</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main444-10-intra</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main12</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main12-intra</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main422-12</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main422-12-intra</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main444-12</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>main444-12-intra</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item row="1" column="0" colspan="3">
          <widget class="QGroupBox" name="groupBox_3">
           <property name="title">
            <string>Quality Goal Metric:</string>
           </property>
           <property name="checkable">
            <bool>false</bool>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="1" column="5">
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>54</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="0" column="0">
             <widget class="QRadioButton" name="losslessEncoding">
              <property name="text">
               <string>Lossless Encoding?</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QRadioButton" name="useAvgBitrate">
              <property name="text">
               <string>Average Bitrate</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QRadioButton" name="useCRF">
              <property name="text">
               <string>CRF Value:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="7" rowspan="2">
             <spacer name="horizontalSpacer_5">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="0" column="3">
             <widget class="QDoubleSpinBox" name="explicitCRF">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="suffix">
               <string> RF</string>
              </property>
              <property name="decimals">
               <number>1</number>
              </property>
              <property name="maximum">
               <double>51.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.500000000000000</double>
              </property>
             </widget>
            </item>
            <item row="0" column="5" colspan="2">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>(Lower is Better Quality)</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="0" column="0" colspan="2">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Default settings for quality are used to maintain quality during the transcoding</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0" colspan="3">
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QCheckBox" name="usePreset">
             <property name="text">
              <string>Preset:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="preset">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <item>
              <property name="text">
               <string>Ultrafast</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Superfast</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Very Fast</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Faster</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Fast</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Medium - Default</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Slow</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Slower</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Very Slow</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_6">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item row="6" column="0" colspan="3">
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QCheckBox" name="useTune">
             <property name="text">
              <string>Tune:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="tune">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <item>
              <property name="text">
               <string>Film - Use for High Quality movie content; lowers deblocking</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Animation - Good for Cartoons; uses higher deblocking and more reference frames</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Grain - Preserves the grain structure in old, grainy fim material</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Still Image - Good for slideshow-like content</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Fast Decode - Allows faster decoding by disabling certain filters</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Zero Latency - Good for fast encoding and low-latency streaming</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_7">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer_7">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>231</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>NSABUtils::CLineEditWithSuffix</class>
   <extends>QLineEdit</extends>
   <header>SABUtils/LineEditWithSuffix.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>transcodeVideo</tabstop>
  <tabstop>onlyTranscodeVideoOnFormatChange</tabstop>
  <tabstop>amdGPUTranscode</tabstop>
  <tabstop>intelGPUTranscode</tabstop>
  <tabstop>nvidiaGPUTranscode</tabstop>
  <tabstop>softwareTranscode</tabstop>
  <tabstop>hwAccel</tabstop>
  <tabstop>videoCodec</tabstop>
  <tabstop>generateNon4kVideo</tabstop>
  <tabstop>average4kBitRate</tabstop>
  <tabstop>averageHDBitRate</tabstop>
  <tabstop>averageSubHDBitRate</tabstop>
  <tabstop>generateLowBitrateVideo</tabstop>
  <tabstop>bitrateThreshold</tabstop>
  <tabstop>losslessEncoding</tabstop>
  <tabstop>useAvgBitrate</tabstop>
  <tabstop>useCRF</tabstop>
  <tabstop>explicitCRF</tabstop>
  <tabstop>usePreset</tabstop>
  <tabstop>preset</tabstop>
  <tabstop>useTune</tabstop>
  <tabstop>tune</tabstop>
  <tabstop>useProfile</tabstop>
  <tabstop>profile</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
